notebooks/community/vertex_endpoints/optimized_tensorflow_runtime/bert_optimized_online_prediction.ipynb

```diff
@@ -134,6 +134,7 @@
         "* Python 3\n",
```

Comment:
<pre class="ql-syntax" spellcheck="false">Line #1.    !gsutil rm -r $BUCKET_URI/*
</pre><p>Upon further reflection, I think we should avoid have a step that deletes everything in the customer GCS bucket. Since buckets have globally unique names I tend to reused buckets for multiple purposes, and I suspect other people do that too.</p><p> </p><p>To avoid that we can copy everything into a specially named subdirectory like</p><p> </p><p>gsutil cp -r <span class="MathJax_Preview" style="color: inherit;">LOCALDIRECTORYFULL/∗</span>BUCKET_URI/vertex_bert_sample/</p><p> </p><p>If we do this we will also need to update all of the model upload commands.</p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/481/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='74'/>

Reply:
For background I added that line because there is a difference when copy to the folder that has something in it and when it doesn't, and after the /∗ change it should not longer be necessary. It is ok to keep it here since the bucket was created in this notebook specifically for the purposes of training this model, and the expectation that it should be deleted in the end to avoid charges.

Reply:
Another option we've used is to provide a boolean `delete_bucket = False` and then allow the user to set it to True if they want to explicitly delete the bucket.

If we do that, it's important we keep the check to see if we're running in a CI environment, to prevent lots of buckets from piling up from our automated tests, e.g.

```python
if delete_bucket or os.getenv("IS_TESTING"):
    ! gsutil -m rm -r $BUCKET_URI
```